modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/transformers/ActionsComponentPropsTransformer.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {render} from 'frontend-js-react-web';
+import React from 'react';
+
+import SidebarPanel from '../SidebarPanel';
+
+export default function propsTransformer({
+	items,
+	sidebarContainerSelector,
+	...otherProps
+}) {
+	const sidebarRefPanel = React.createRef();
+
+	const actions = {
+		showInfo() {
+			showSidebar();
+		},
+		showMetrics() {
+			showSidebar();
+		},
+	};
+
+	const hideSidebar = () => {
+		sidebarRefPanel.current.close();
+	};
+
+	const setSidebarPanelRef = (element) => {
+		sidebarRefPanel.current = element;
+	};
+
+	const showSidebar = () => {
+		if (!sidebarRefPanel.current) {
+			render(
+				SidebarPanel,
+				{
+					onClose: hideSidebar,
+					ref: setSidebarPanelRef,
```

Comment:
Instead of passing a function, can't you just pass the ref directly?

Reply:
Probably I can but in the next pr I'll be doing this:

```
const _setSidebarPanel = (element) => {
		Liferay.component(`${namespace}sidebar`, element);
	};
```

I'm registering the component so it's available on every `ActionsComponentPropsTransformer` instance so we can use the same one just changing the `view` or the `fetchURL`